stock_analytic/models/stock.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2013 Julius Network Solutions
+# Copyright 2015 Clear Corp
+# Copyright 2016 OpenSynergy Indonesia
+# Copyright 2017 ForgeFlow S.L.
+# Copyright 2018 Hibou Corp.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class StockMove(models.Model):
+    _name = "stock.move"
+    _inherit = ["stock.move", "analytic.mixin"]
+
+    def _prepare_account_move_line(
+        self, qty, cost, credit_account_id, debit_account_id, svl_id, description
+    ):
+        self.ensure_one()
+        res = super(StockMove, self)._prepare_account_move_line(
+            qty, cost, credit_account_id, debit_account_id, svl_id, description
+        )
+        for line in res:
+            if (
+                line[2]["account_id"]
+                != self.product_id.categ_id.property_stock_valuation_account_id.id
+            ):
+                # Add analytic account in debit line
+                if self.analytic_distribution:
+                    line[2].update(
+                        {"analytic_distribution": self.analytic_distribution}
+                    )
```

Comment:
```suggestion
        if not self.analytic_distribution:
            return res
        for line in res:
            if (
                line[2]["account_id"]
                != self.product_id.categ_id.property_stock_valuation_account_id.id
            ):
                # Add analytic account in debit line
                line[2].update(
                    {"analytic_distribution": self.analytic_distribution}
                )
```
Just to make it slightly more efficient.